src/core/ext/filters/client_channel/lb_policy/xds/xds_cluster_resolver.cc

```diff
@@ -510,6 +489,35 @@ void XdsClusterResolverLb::LogicalDNSDiscoveryMechanism::ResolverResultHandler::
       discovery_mechanism_->index(), std::move(update));
 }
 
+//
+// XdsClusterResolverLb::DiscoveryMechanismEntry
+//
+
+const XdsClusterResolverLbConfig::DiscoveryMechanism&
+XdsClusterResolverLb::DiscoveryMechanismEntry::config() const {
+  return discovery_mechanism->parent()
+      ->config_->discovery_mechanisms()[discovery_mechanism->index()];
+}
+
+std::string XdsClusterResolverLb::DiscoveryMechanismEntry::GetChildPolicyName(
+    size_t priority) const {
+  const size_t child_number = priority_child_numbers[priority];
+  const auto& discovery_config = config();
+  absl::InlinedVector<absl::string_view, 9> parts = {
+      "{cluster=", discovery_config.cluster_name, ","};
```

Comment:
shouldn't the cluster name by itself be enough to differentiate amongst these priorities?

Reply:
Good point.  Done.

Reply:
Followup question, what happens if we get an update for an EDS cluster which changes the EDS service name or the load reporting service name?

Not sure if that's possible in practice. But if it was, could that re-use a priority and so crash in the `XdsClusterImpl::Update`?

Reply:
The EDS service name is set in the CDS resource, not the EDS resource -- it's basically the way that the CDS resource tells the client which EDS resource to use.  And it is certainly possible to get a CDS update that changes the EDS service name.

The config of the xds_cluster_resolver policy includes a list of discovery mechanisms, and each discovery mechanism includes the [EDS service name](https://github.com/grpc/grpc-proto/blob/ab96cf12ec7ce135e03d6ea91d96213fa4cb02af/grpc/service_config/service_config.proto#L399) and [LRS server name](https://github.com/grpc/grpc-proto/blob/ab96cf12ec7ce135e03d6ea91d96213fa4cb02af/grpc/service_config/service_config.proto#L383).  When the xds_cluster_resolver policy gets a config update that changes the list of discovery mechanisms, it will [gracefully swap itself out](https://github.com/grpc/grpc/blob/8f3cd544cdd95092d514ae30bb67a037132d2e35/src/core/ext/filters/client_channel/lb_policy/xds/xds_cluster_resolver.cc#L1240) -- i.e., instead of updating the existing instance, it will create a whole new instance of the policy, and it will do a graceful swap over to that new instance.